addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3845,6 +3836,22 @@ var SnippetsMenu = Widget.extend({
     _allowInTranslationMode($snippet) {
         return globalSelector.is($snippet, { onlyTextOptions: true });
     },
+    /**
+     * Allows to update the snippets to build & adapt dynamic content.
+     *
+     * @private
+     */
+    _adaptSnippetContent($target) {
```

Comment:
Kinda confusing with `callPostSnippetDrop` and `onBuilt`, no better name possible?

The `o_snippet_drop_in_only` is a very specific cas: we don't want to build snippet option so "before the drop", we have to do something. But for the highlight case, I am not sure why it could not be "post drop" (it seems fine either way?).

Reply:
> Kinda confusing with callPostSnippetDrop and onBuilt, no better name possible?

`_updateDroppedSnippet()` ?

> The o_snippet_drop_in_only is a very specific cas: we don't want to build snippet option so "before the drop", we have to do something. But for the highlight case, I am not sure why it could not be "post drop" (it seems fine either way?).

Probably I'm missing something here, but from what I see in the code, we are in a "post drop" situation :thinking:, since after [this part of the code](https://github.com/odoo/odoo/pull/182208/files#diff-52a4f9d2c217548e69e6b7fd097f286f1754a6389734eea254b87255e501cbefR3588-R3594), the snippet should be dropped and available in the "DOM" ?

In general, highlights need the content to be "fully loaded" to compute sizes correctly… But here I don't think it's an issue (if the content is resized because of external factors: font applied,... The public widget should take care of it :+1:).

Remark: In a first implementation, I simply triggered `_refreshPublicWidgets()` `onBuilt()`, but I removed it because of the highlights' application delay :grimacing: :

[17.0-d&d-snippet-highlights.webm](https://github.com/user-attachments/assets/3f646bea-41b8-47cc-b859-19c7a26ef1f5)

Reply:
> Probably I'm missing something here, but from what I see in the code, we are in a "post drop" situation 🤔

Of course. What I meant was that we have a "post-drop" function already and you are just creating a new one. So we now would have two post-drop callbacks.

> _updateDroppedSnippet() ?

I don't know, I'll re-review and merge